k2/csrc/util.h

```diff
@@ -0,0 +1,24 @@
+// k2/csrc/util.h
+
+// Copyright (c)  2020  Haowen Qiu
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#ifndef K2_CSRC_UTIL_H_
+#define K2_CSRC_UTIL_H_
+
+#include <functional>
+
+#include "k2/csrc/fsa.h"
+
+namespace k2 {
+
+// boost::hash_combine
+template <class T>
+inline void hash_combine(std::size_t *seed, const T &v) {
```

Comment:
To follow the google c++ style, the output paremter should be the last one.

So the siganature should be
```cpp
inline void hash_combine(const T &v, std::size_t *seed)
```

Reply:
I think it's better here to put `seed` as the first parameter since: 

1. it it the input prarameter
2. when we read the code from left to right, we see first `seed` then `value` which is more readable.
3. consistent siganature with `boost::hash_combine`

Reply:
@danpovey may have something to say about this.

Reply:
I think I'm OK to have seed first in this instance, specifically because it's following the boost::hash_combine tradition.
BTW, it's a bit annoying that I can't search for conversations in github because they are all resolved.  I dont know if there's a way to mark resolved without the text disappearing...